src/server/game/Craft/Crafting.cpp

```diff
@@ -0,0 +1,243 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Crafting.h"
+#include "Containers.h"
+#include "DB2Stores.h"
+#include "DB2Structure.h"
+#include "Item.h"
+#include "ItemBonusMgr.h"
+#include "SharedDefines.h"
+#include "Spell.h"
+#include "SpellInfo.h"
+#include "SpellMgr.h"
+#include "SpellPackets.h"
+#include "Player.h"
+
+SpellCastResult Crafting::DoCraft(uint32 craftingDataId)
```

Comment:
You're mixing responsibilities like validation and computation. Given that this system is very tedious, I would break it down into smaller helper functions:

ValidateReagents
Determine/ComputeCraftQuality
ConsumeReagents